vmaas_sync/vmaas_sync.py

```diff
@@ -385,11 +385,26 @@ def _get_updated_repos(self, conn) -> list:
                 LOGGER.info("%s systems scheduled for re-evaluation", total_scheduled)
             conn.commit()
 
+    def _send_websocket_message(self, message):
+        if self.vmaas_websocket:
+            asyncio.ensure_future(self.vmaas_websocket.write_message(message))
+        else:
+            self._websocket_reconnect()
+            asyncio.ensure_future(self.vmaas_websocket.write_message(message))
+
     def _read_websocket_message(self, message):
         """Read incoming websocket messages."""
         future = None
         if message is not None:
+            if message == "webapps-refreshing":
+                LOGGER.info("Stopping kafka producer while refreshing VMaaS..")
+                self.evaluator_queue.vmaas_refreshing = True
```

Comment:
Are you sure this works? Isn't evaluator queue used in completely different process in separate container? How is this flag going to get there? Evaluator queue in this context is only the ```vulnerability.evaluator.recalc``` topic.

Reply:
not `vulnerability.evaluator.upload` topic here? `self.evaluator_queue = mqueue.MQWriter(mqueue.EVALUATOR_TOPIC)`
When I tested, it seems that It was working

Reply:
```mqueue.EVALUTOR_TOPIC``` is defined as ```EVALUATOR_TOPIC = os.getenv('MESSAGE_TOPIC', 'vulnerability.evaluator.upload')``` and ```MESSAGE_TOPIC``` env variable is defined as this for vmaas_sync component:

https://github.com/RedHatInsights/e2e-deploy/blob/master/templates/vulnerability/vulnerability-engine-vmaas-sync.yml#L111

but even if it was correct, how does this flag get into other containers, it may be shared code but the runtime is not shared

Reply:
I realized I will try to come up with something. Do you have any suggestions on how to deal with it?

Reply:
As I'm thinking more about it I'd not do anything complicated message sending like this. What about making vmaas-webapp returning HTTP 503 while the refresh is in progress and evaluator/vmaas-sync make retrying until something else than 503 is returned?